src/config-schema.coffee

```diff
@@ -155,32 +172,36 @@ module.exports =
       softTabs:
         type: 'boolean'
         default: true
+        description: 'If the `Tab Type` config setting is set to "auto" and autodetection of tab type from buffer content fails, then this config setting determines whether a soft tab or a hard tab will be inserted when the Tab key is pressed.'
```

Comment:
> Yeah, if there's no content, the auto detect doesnt work, and will fall back to this setting.

@benogle Just to make sure I'm following along -- can you share a link to the code where this behavior is implemented? Here's what I'm seeing:

1. Auto-detect kicks in [here](https://github.com/atom/atom/blob/3636eb35d8a7dbdc00cfbdaba7280c26abb08ca2/src/text-editor.coffee#L2387-L2388) if the tabType option is set to "auto". If auto-detect fails, then the defaultValue is used (what's this?) or the value of the softTabs config setting.
2. Auto-detect is implemented [here](https://github.com/atom/atom/blob/3636eb35d8a7dbdc00cfbdaba7280c26abb08ca2/src/text-editor.coffee#L2356-L2364) and fails for buffers which have no non-comment lines with leading whitespace (not necessarily an empty file).

I'm not sure what this defaultValue is, at which moments is the tab type of a buffer determined (When the file is first opened in Atom? Every time it's opened? Every time it's changed?), and how does the ability to manually change the tab type for a single buffer come affect everything (using the `Editor: Toggle Soft Tabs` command)?